### ISSUE TITLE
Gifox fails to create animated gif

### ISSUE THREAD

**Issue by pct196** [Gifox/Tracker#137]
I've encountered an issue today where Gifox is failing to create an animated gif. I've attached an image showing the Gifox progress bar. I took the screen recording at around 08:12 today and it is now 18:56.

Previously I had an issue where I removed 20 seconds from a 60 second gif and encountered the same issue.

![CleanShot 2020-03-11 at 18 53 42](https://user-images.githubusercontent.com/3455606/76394483-b4232580-63c9-11ea-9456-44780204255f.png)

Please let me know what information you would like me to provide from here.

**Comment by 2color** [comment]
I'm experiencing the same problem

**Comment by ianbytchek** [comment]
Hey fellas, thanks for reporting this and apologies for the laggy response. Is this the original or version 2 are we talking about?

**Comment by pct196** [comment]
I'm running v2.1.1.

**Comment by 2color** [comment]
Same here

**Comment by ianbytchek** [comment]
Thanks for confirming. A couple of things.

1. Today version 2.1.2 will be released, I'll ping here when it's ready. It is unlikely to fix the issue, but includes improved diagnostic tools and hopefully will provide the information for understanding what's going wrong.

2. In the meantime, please check if restarting your Mac helps. In 2.1.0 we had a lot of issues with standard H.264 encoder, which would fail with larger recordings depending on the hardware. In 2.1.1 we replaced it with a newer alternative and it was supposed to solve such issues. It looks like we're back to square one… I see that you have some GIFs, so I'm assuming this might be size-duration related. Can you try figuring out if it's failing when creating a GIF above certain size? Start with a small & short recording and continue increasing the size?

**Comment by ianbytchek** [comment]
@pct196, @2color 2.1.2 just got published.

**Comment by pct196** [comment]
@ianbytchek I purchased Gifox through the mac app store. When apple approves the latest update I will install and test.

**Comment by pct196** [comment]
@ianbytchek I did some testing today without any issues. Can you send me an email address where I can send the device logs please?

# Tests done

Tests with no scaling (Internal GPU monitor):

- 10s - ✅
- 20s - ✅
- 30s - ✅
- 60s - ✅

Tests with scaling (Internal GPU monitor):

- 10s - ✅
- 20s - ✅
- 30s - ✅
- 60s - ✅

Tests with no scaling (External GPU monitor):

- 60s - ✅

Tests with scaling (External GPU monitor):

- 60s - ✅

**Comment by ianbytchek** [comment]
@pct196 Oh wow, thanks for the thorough breakdown! Please send everything to team@gifox.io, though if it didn't fail with 2.1.2 the logs probably won't be of much use, but I'll have a peak to see if there's anything standing out.

**Comment by lucho-dev** [comment]
This started to happen to me a couple of days ago after I changed the default export settings. The gif I was working on freezed and any new gif I tried to make (even with the settings defaulted) fails to appear. Progress bar get stuck at 100%
<img width="296" alt="Screen Shot 2020-03-29 at 18 56 42" src="https://user-images.githubusercontent.com/13853441/77862123-d0a7c600-71ef-11ea-855a-f8fe0533c0cd.png">
I use version 2.1.2. Here are the app logs and current settings. 
[App Logs.txt](https://github.com/Gifox/Tracker/files/4399874/App.Logs.txt)
[App Settings.txt](https://github.com/Gifox/Tracker/files/4399867/App.Settings.txt)

**Comment by pct196** [comment]
@ianbytchek, I haven't had any issues since v2.1.2.

If the issue @lucho-dev experienced has been fixed in a later version I'd be happy to close off this issue.

**Comment by ianbytchek** [comment]
It must have been a random glitch that caused it in the first place. The good news it seems like we've finally sorted out #135 and in the upcoming beta will include a proper work-around to address this issue. Let's keep it open for now, it the meanwhile, if you can enable pre-releases in general preferences and test the next beta it would be of great help!

**Comment by x0000ff** [comment]
I still have this problem :( ( Gifox 2.1.4 )
Some gifs just freeze:
<img width="338" alt="Screen Shot 2020-04-23 at 15 33 14" src="https://user-images.githubusercontent.com/1062240/80141590-c4f1ba00-8577-11ea-8565-1554dd61b610.png">

I can't reproduce it 100%, I started facing this problem a week ago I think 🤔

**Comment by ianbytchek** [comment]
Friends, Gifox 2.2.0 Beta was just published and available for update, make sure to enable prerelease updates in general preferences. It provides a proper solution for this issue amongst other improvements and fixes. I need your input to confirm it works as expected and there are no regressions. The next public release is going to be awesome! 👍

**Comment by x0000ff** [comment]
Thanks, @ianbytchek! I'll try it right now 🙌

**Comment by pct196** [comment]
@ianbytchek, I've been running the latest beta for a couple of day's without any video encoding issues. I'll continue to test as I need to make videos.

**Comment by matthewrkula** [comment]
I am using Gifox v2.2.1 and still seeing this issue periodically. Killing Gifox and reopening seems to fix it for me, but subsequent gif creation seems to fail.

**Comment by ianbytchek** [comment]
From latest feedback and observations it's most likely to be #159. Can you confirm if this happens between relaunching and Mac going into sleep? Or does it simply eventually return even though after relaunch the app works as normal for sometime?

**Comment by ianbytchek** [comment]
Closing this in favor of #159. This used to happen due to loss of contact between the main app and media service. A few versions back we found away to avoid using media service all together, so, technically, the original issue should be resolved. 

While fighting with #159 we increased per-frame processing time to 5 seconds to handle very rare system lags, which in certain situations would result all frames failing to render with 5 second pause. As the result it would probably look exactly like this issue. In 2.2.2 release we reverted to simply failing the export as soon as any frame times out to process as the previous approach clearly didn't work out… 😅
